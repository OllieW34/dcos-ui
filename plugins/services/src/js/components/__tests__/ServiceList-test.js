/* eslint-disable no-unused-vars */
const React = require("react");
/* eslint-enable no-unused-vars */
const ReactDOM = require("react-dom");

const renderer = require("react-test-renderer");

const ServiceList = require("../ServiceList");
const ServiceTree = require("../../structs/ServiceTree");

let thisContainer, thisInstance;

describe("ServiceList", function() {
  describe("#shouldComponentUpdate", function() {
    beforeEach(function() {
      var services = new ServiceTree({ items: [{ name: "foo" }] });
      thisContainer = global.document.createElement("div");
      thisInstance = ReactDOM.render(
        <ServiceList services={services.getServices().getItems()} />,
        thisContainer
      );
    });

    afterEach(function() {
      ReactDOM.unmountComponentAtNode(thisContainer);
    });

    it("allows update", function() {
      var shouldUpdate = thisInstance.shouldComponentUpdate({ a: 1 });
      expect(shouldUpdate).toEqual(true);
    });

    it("does not allow update", function() {
      var shouldUpdate = thisInstance.shouldComponentUpdate(thisInstance.props);
      expect(shouldUpdate).toEqual(false);
    });
  });

  describe("#getServices", function() {
    const services = new ServiceTree({
      items: [
        {
          name: "foo",
          tasksStaged: 0,
          tasksRunning: 0,
          tasksHealthy: 0,
          tasksUnhealthy: 0,
          instances: 0,
          deployments: []
        }
      ]
    });

    beforeEach(function() {
      thisContainer = global.document.createElement("div");
      thisInstance = ReactDOM.render(
        <ServiceList services={services.getServices().getItems()} />,
        thisContainer
      );
    });

    afterEach(function() {
      ReactDOM.unmountComponentAtNode(thisContainer);
    });

    it("returns services that have a value of two elements", function() {
      const component = renderer.create(
        <ServiceList services={services.getServices().getItems()} />
      );
      const tree = component.toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
