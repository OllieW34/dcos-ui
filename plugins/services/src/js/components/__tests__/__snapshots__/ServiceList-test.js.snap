// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceList #getServices returns services that have a value of two elements 1`] = `
<div
  className="dashboard-health-list"
>
  <ul
    className="list list-unstyled"
  >
    <li
      className="list-item"
    >
      <span
        className="dashboard-health-list-item-description text-overflow"
      >
        <a
          className="dashboard-health-list-item-cell emphasis"
          href={null}
          onClick={[Function]}
        >
          
        </a>
      </span>
      <div
        className="dashboard-health-list-health-label"
      >
        <div
          className="tooltip-wrapper"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <svg
            className="icon icon-light-grey icon-mini"
          >
            <use
              xlinkHref="#icon-system--circle-minus"
            />
          </svg>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
